lib/web_ui/lib/src/engine/skwasm/skwasm_impl/renderer.dart

```diff
@@ -369,11 +369,12 @@ class SkwasmRenderer implements Renderer {
     if (contentType == null) {
       throw Exception('Could not determine content type of image from data');
     }
-    final ui.Codec baseDecoder = SkwasmImageDecoder(
+    final SkwasmImageDecoder baseDecoder = SkwasmImageDecoder(
       contentType: contentType,
       dataSource: list.toJS,
       debugSource: 'encoded image bytes',
     );
+    await baseDecoder.initialize();
```

Comment:
Not sure how the error manifested on the failing tests, but it may make sense to spit out a warning on the `SkwasmImageDecoder` if users attempt to call method instances before they call `initialize()`? It doesn't look like an optional method to call, right?

Reply:
Yeah, the whole point of this change is that we should never hand the codec back to the user in an uninitialized state. Particularly because the `ui.Codec` API doesn't have the `initialize` method on it. So I don't think there is any reason to have warnings, vending an uninitialized codec is an error of the engine.